### DISCUSSION TITLE
应该如何进行文件跳转?

### DISCUSSION THREAD

**Discussion by Linya-IronMan** [nshen/learn-neovim-lua#101] [Q&A]
特别是进行前端的文件跳转的时候，webpack 中会进行一些resolve设置
比如 "@": "./src" 
会将 @ 符号指向 src 目录，这样就完全无法跳转到对应的引入文件中.

---

在写vue 文件的时候，会遇到 `this.num` 这样访问变量的情况，这个时候也不知道应该怎么跳转到num 定义的地方。
gd 好像就只是在全局搜索文本而已

---

还有一些快捷键,  gj , gk , gr 都是什么意思？
我在 lspsaga 上找到了原始的内容，但是 ...  diagnostic_jump 是什么意思？我尝试过，但是跳转后的内容和跳转前的内容看不出有什么关联性，并且在vue中还是无法跳转。
<img width="788" alt="image" src="https://user-images.githubusercontent.com/77786952/185857055-b42b8405-f1d0-4012-9553-57fe1efc9816.png">

---

快捷键 gh 应该怎么使用？这个应该是模拟鼠标悬停的效果，可以查看一个变量的类型或者javadoc这样的内容，但是在 vue 文件中同样失效。

学习vim快三个月了，还是因为各种问题，无法用到实际生产中，快要寄了

**Comment by nshen** [comment]
vscode 也不可能认得 webpack resolve 里的自定义的路径吧? 认得吗？

我觉得文件跳转应该是 language server 提供的，typescript 需要 tsconfig 里的 paths 提供，类似这样

```
  "compilerOptions": {
    "paths": {
      "@/*": ["./src/*"]
    }
  }
```

你的所有功能都不好用是因为lsp没配好，我是写 react 的，配置上tsserver就可以了，.js .tsx就都可用了。vue 是不是都必须写在自定义扩展名的文件里啊？tsserver 可能不能直接支持 vue 的 filetype，你还要配置一下 vue 的 language server之类的。

**Comment by allworldg** [comment]
看不懂的可以：1.中文翻译 2. 看插件文档 3. 看官方文档 4. google关键字  5. reddit或so 6. 实在顶不住就VSCode
